### PR TITLE
some Apollo experiments

### PR DESCRIPTION
**Pull request by jasongornall**
## App preview:

- [Web](https://sg-web-jg-experimenting.onrender.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/components/FilteredConnection/hooks/useConnection.ts [hunk 1]
```diff
@@ -153,15 +159,23 @@ export const useConnection = <TResult, TVariables, TData>({
...
     }
 
     const refetchAll = useCallback(async (): Promise<void> => {
         const first = connection?.nodes.length || firstReference.current.actual
+        console.log('REFETCH', {
+            ...variables,
+            first,
+        })
+        if (isFetchingMore.current) {
+            console.log('SKIPPED')
```
Comment (jasongornall) [inline client/web/src/components/FilteredConnection/hooks/useConnection.ts:175]: this is the core fix.. if we are fetching and there is a pending network call to load more we skip over it

File: client/web/src/components/FilteredConnection/hooks/useConnection.ts [hunk 2]
```diff
@@ -173,11 +188,28 @@ export const useConnection = <TResult, TVariables, TData>({
...
+    } else if (isFetchingMore.current) {
+        isLoading = true;
+    }
     return {
         connection,
-        loading,
+        loading: isLoading,
         error,
-        fetchMore: fetchMoreData,
+        fetchMore: () => {
```
[the remark this answers is not in the document]
Comment (jasongornall) [reply client/web/src/components/FilteredConnection/hooks/useConnection.ts:203]: https://github.com/apollographql/apollo-client/issues/6354 I still need to investigate a bit.

[the remark this answers is not in the document]
Comment (jasongornall) [reply client/web/src/components/FilteredConnection/hooks/useConnection.ts:203]: notifyOnNetworkStatusChange: true, // Ensures loading state is updated on `fetchMore`

I see this in the code but not seeing it get reflected.

File: client/web/src/components/FilteredConnection/hooks/useConnection.ts (hunk 1 above)
Comment (jasongornall) [reply client/web/src/components/FilteredConnection/hooks/useConnection.ts:176]: alternatively I could probably useCallback the setInterval so it clears out and starts a new one

**Comment by eseliger** [comment]
As part of #wg-delete-week I am going to close this PR. If it’s still needed, feel free to reopen and restore the branch.

**Pull request closed without merging**